src/sentry/snuba/query_subscription_consumer.py

```diff
@@ -46,22 +217,77 @@ class InvalidSchemaError(InvalidMessageError):
     pass
 
 
+class QuerySubscriptionStrategyFactory(ProcessingStrategyFactory[KafkaPayload]):
+    def __init__(self, topic: str):
+        self.topic = topic
+
+    def create_with_partitions(
+        self,
+        commit: Commit,
+        partitions: Mapping[Partition, int],
+    ) -> ProcessingStrategy[KafkaPayload]:
+        def process_message(message: Message[KafkaPayload]) -> None:
+            with sentry_sdk.start_transaction(
+                op="handle_message",
+                name="query_subscription_consumer_process_message",
+                sampled=random() <= options.get("subscriptions-query.sample-rate"),
+            ), metrics.timer("snuba_query_subscriber.handle_message"):
+                value = message.value
+                assert isinstance(value, BrokerValue)
+                offset = value.offset
+                partition = value.partition.index
+                try:
+                    handle_message(
+                        str(message.payload.value),
```

Comment:
Why did you need to cast the bytes to a string?

Reply:
Hmm at the time I was getting a JSON decode error but it seems to run fine w/o it now, updated.